StringPrototypeExtension.js

```diff
@@ -0,0 +1,185 @@
+String.prototype.hasVowels = function() {
+  var vowelRegex = /[aeiou]/gi;
+  var hasVowel = vowelRegex.test(this);
+  return hasVowel;
+};
+
+String.prototype.isQuestion = function() {
+  var isQuestionRegex = /\w+[?]$/;
+  return isQuestionRegex.test(this);
+};
+
+String.prototype.numberWords = function() {
+  var numberWordRegex = [
+    "zero",
+    "one",
+    "two",
+    "three",
+    "four",
+    "five",
+    "six",
+    "seven",
+    "eight",
+    "nine"
+  ];
+  var digitsArray = this.match(/\d/g);
+  var numberToWord = "";
+  digitsArray.forEach(digit => {
+    numberToWord += numberWordRegex[digit].concat(" ");
+  });
+  return numberToWord;
+};
+
+String.prototype.wordsToArray = function() {
+  var wordRegex = /\b[0-9a-zA-Z_\-?]+\b/g;
+  var wordArray = this.match(wordRegex);
+
+  return wordArray;
+};
+
+String.prototype.wordCount = function() {
+  return this.wordsToArray().length;
+};
+
+String.prototype.isDigit = function() {
+  var digitRegex = /\d/g;
+
+  if (!digitRegex.test(this)) {
+    return false;
+  }
+  return this.match(digitRegex).length === 1 ? true : false;
+};
+
+String.prototype.toCurrency = function() {
+  var regexPattern = /^\d+(\.\d+)?$/g;
+  if (!regexPattern.test(this)) {
+    return false;
+  }
+  var currency = Number(this)
+    .toFixed(2)
+    .replace(/\d(?=(\d{3})+\.)/g, "$&,");
+
+  return currency;
+};
+
+String.prototype.fromCurrency = function() {
+  var regex = /(\d+),(\d+),?(\d+)?(\.)?(\d+)?/;
+  var num = this.replace(regex, "$1$2$3$4$5");
+  return parseFloat(num);
+};
+
+String.prototype.toUpper = function() {
+  var regex = /.+/g;
+  var upperCasedString = "";
+  if (regex.test(this)) {
+    for (var character of this) {
+      var asciiValue = character.charCodeAt();
+      if (asciiValue >= 97 && asciiValue <= 122) {
+        upperCasedString += String.fromCharCode(asciiValue - 32);
+        continue;
+      }
+      upperCasedString += character;
+    }
+  } else {
+    return false;
+  }
+
+  return upperCasedString;
+};
+
+String.prototype.toLower = function() {
+  var regex = /.+/g;
+  var lowerCasedString = "";
+  if (regex.test(this)) {
+    for (var character of this) {
+      var asciiValue = character.charCodeAt();
+      if (asciiValue >= 60 && asciiValue <= 90) {
+        lowerCasedString += String.fromCharCode(asciiValue + 32);
+        continue;
+      }
+      lowerCasedString += character;
+    }
+  } else {
+    return false;
+  }
+
+  return lowerCasedString;
+};
+
+String.prototype.ucFirst = function() {
+  var regex = /\b[a-zA-Z_\-0-9]+[\d\+]?\b/g;
+  var upperCasedFirstString = false;
+
+  if (regex.test(this)) {
```

Comment:
Please check for the NEGATIVE condition first; this way, you would get rid of the else logic